### PR TITLE
Dev mpatch 1

### PR DESCRIPTION
**Pull request by DebajyotiSahoo13**

### CODE REVIEW

File: src/model/plots/agglomerative_clustering_3d.png
Comment (lokeshmohanty) [inline src/model/plots/agglomerative_clustering_3d.png:1]: Plots should not be tracked by git

File: src/data/data_loader.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def __init__(self, n_samples: int = 1000):
+        self.data = self.generate_radar_data(n_samples)
+
+    def __len__(self) -> int:
+        return len(self.data)
+
+    def __getitem__(self, idx: int) -> torch.Tensor:
+        return torch.tensor(self.data.iloc[idx].values, dtype=torch.float32)
+
+    @staticmethod
```
Comment (lokeshmohanty) [inline src/data/data_loader.py:20]: Any reason for making these methods static?

File: src/data/data_loader.py
Comment (lokeshmohanty) [inline src/data/data_loader.py:1]: The file name is too generic, you can change it something like `radar-synthetic.py`

**Review by lokeshmohanty — CHANGES_REQUESTED** [review]
Good work! Make the suggested changes.

You will have to add tests cases for all these as currently it is not obvious if there are any problems with the code, though that can be done in a separate pull request after this is merged.

**Comment by lokeshmohanty** [comment]
Change the commit messages as well. The current one's are not very helpful. You can merge all changes into a single commit with message "add common clustering algorithms" with proper description.

**Comment by lokeshmohanty** [comment]
You can install matplotlib by using `poetry add matplotlib`. That will add it to poetry.lock file and pyproject.toml

**Comment by DebajyotiSahoo13** [comment]
@lokeshmohanty  Cleared all test checks.Please review and suggest changes.
I will rebase all the git commit messages at the very end.
Let me know if any change in code structure.

File: src/data/data_loader.py
Comment (lokeshmohanty) [reply src/data/data_loader.py:1]: You duplicated this file. Delete it.

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
+# """Main script for running clustering models."""
+# import os
```
Comment (lokeshmohanty) [inline src/main.py:2]: Remove these comments

File: .gitignore [hunk 3]
```diff
@@ -1 +1,3 @@
 **/__pycache__
+# Ignore generated plots
+/src/plots/
```
Comment (lokeshmohanty) [inline .gitignore:3]: Don't add `/src/plots` in gitignore. All the files to be ignored should be put outside `src` folder

**Review by lokeshmohanty — COMMENTED** [review]
- `.dvc` directory has lot of duplicate files. Remove all the duplicates.
- If you intend to rename `data` folder to `dataset`, don't just duplicate it and name it `dataset`, you need to remove the `data` folder as well.

**Comment by DebajyotiSahoo13** [comment]
@lokeshmohanty Made the changes, do let me know if any other changes

**Pull request closed without merging**